packages/recommend-core/src/personalization/__tests__/index.test.ts

```diff
@@ -0,0 +1,140 @@
+import * as affinities from '../getAffinities';
+import * as strategy from '../getStrategy';
+import { getPersonalizationFilters } from '../index';
+
+describe('personalization', () => {
+  describe('getPersonalizationFilters', () => {
+    it('should return an empty array if userToken parameter is missing', async () => {
+      jest.spyOn(console, 'warn').mockImplementation();
+
+      const result = await getPersonalizationFilters({
+        userToken: '',
+        region: 'eu',
+        apiKey: '123465',
+        appId: '123456',
+      });
+
+      // eslint-disable-next-line no-console
+      expect(console.warn).toHaveBeenCalledWith(
+        `[Algolia Recommend] parameter 'userToken' is required to enable personalization.`
+      );
+      expect(result).toEqual([]);
+    });
+
+    it('should return an empty array if missing userToken', async () => {
+      await expect(
+        getPersonalizationFilters({
+          userToken: 'token',
+          region: '',
+          apiKey: '',
+          appId: '',
+        })
+      ).rejects.toThrow(
+        `[Algolia Recommend] parameters 'region', 'apiKey' and 'appId' are required to enable personalization.`
+      );
+    });
```

Comment:
The test name doesn't match its content.

Reply:
good catch